api/v1alpha1/nats_types.go

```diff
@@ -81,6 +63,100 @@ type NATS struct {
 	Status NATSStatus `json:"status,omitempty"`
 }
 
+// NATSStatus defines the observed state of NATS.
+type NATSStatus struct {
+	State      State              `json:"state"`
+	Conditions []metav1.Condition `json:"conditions,omitempty"`
+}
+
+// NATSSpec defines the desired state of NATS.
+type NATSSpec struct {
+	// Cluster defines configurations that are specific to NATS clusters.
+	Cluster `json:"cluster"`
+
+	// JetStream defines configurations that are specific to NATS JetStream.
+	// +optional
+	JetStream `json:"jetStream,omitempty"`
+
+	// JetStream defines configurations that are specific to NATS logging in NATS.
+	// +optional
+	Logging `json:"logging,omitempty"`
+
+	// Resources defines resources for NATS.
+	// +optional
+	Resources `json:"resources,omitempty"`
+
+	// Annotations allows to add annotations to NATS.
+	// +optional
+	Annotations []string `json:"annotations,omitempty"`
+
+	// Labels allows to add Labels to NATS.
+	// +optional
+	Labels []string `json:"labels,omitempty"`
+}
+
+// Cluster defines configurations that are specific to NATS clusters.
+type Cluster struct {
+	// Size of a NATS cluster, i.e. number of NATS nodes.
+	Size int `json:"size"`
+}
+
+// JetStream defines configurations that are specific to NATS JetStream.
+type JetStream struct {
+	// MemStorage defines configurations to memory storage in NATS JetStream.
+	// +optional
+	MemStorage `json:"memStorage,omitempty"`
+
+	// FileStorage defines configurations to file storage in NATS JetStream.
+	// +optional
+	FileStorage `json:"fileStorage,omitempty"`
+}
+
+// MemStorage defines configurations to memory storage in NATS JetStream.
+type MemStorage struct {
+	// Enable allows the enablement of memory storage.
+	Enable bool `json:"enable"`
+
+	// Size defines the mem.
+	Size string `json:"size"`
+}
+
+// FileStorage defines configurations to file storage in NATS JetStream.
+type FileStorage struct {
+	// StorageClassName defines the file storage class name.
+	StorageClassName string `json:"storageClassName"`
+
+	// Size defines the file storage size.
+	Size string `json:"size"`
+}
+
+// Logging defines logging options.
+type Logging struct {
+	// Debug allows debug logging.
+	Debug bool `json:"debug"`
+
+	// Trace allows trace logging.
+	Trace bool `json:"trace"`
+}
+
+type Resources struct {
```

Comment:
Why not re-use `ResourceRequirements` struct from `corev1` pkg? instead of defining our own struct.

Reply:
Because I wasn't aware of its existence.

Reply:
Fixed that.